services/github/github.go

```diff
@@ -735,24 +742,37 @@ func (s *Service) Register(oldService types.Service, client types.MatrixClient)
 	return nil
 }
 
-// defaultRepo returns the default repo for the given room, or an empty string.
-func (s *Service) defaultRepo(roomID id.RoomID) string {
-	logger := log.WithFields(log.Fields{
-		"room_id":     roomID,
-		"bot_user_id": s.ServiceUserID(),
-	})
+func (s *Service) loadBotOptions(roomID id.RoomID, logger *log.Entry) (result map[string]interface{}, err error) {
 	opts, err := database.GetServiceDB().LoadBotOptions(s.ServiceUserID(), roomID)
 	if err != nil {
-		if err != sql.ErrNoRows {
-			logger.WithError(err).Error("Failed to load bot options")
+		if err == sql.ErrNoRows {
+			logger.Info("no bot options")
+			return make(map[string]interface{}), nil
+		} else {
+			err := errors.New("Failed to load bot options")
+			logger.WithError(err).Error(err)
+			return nil, err
 		}
-		return ""
 	}
 	// Expect opts to look like:
 	// { github: { default_repo: $OWNER_REPO } }
 	ghOpts, ok := opts.Options["github"].(map[string]interface{})
 	if !ok {
-		logger.WithField("options", opts.Options).Error("Failed to cast bot options as github options")
+		err = fmt.Errorf("Failed to cast bot options as github options")
+		logger.WithField("options", opts.Options).Error(err)
```

Comment:
I don't think you need this?

Reply:
The rearrangement here is because I refactored the option loading into `loadBotOptions`, and I'd like to return an error to the caller if it occurs, so created an `err` which I can return .. unless you were suggesting something else